### PR TITLE
test: New test group to test for side effects

### PR DESCRIPTION
**Pull request by mgrandis**
*Issue #, if available:*

*Description of changes:*
Added a new group test `test_transform_success_no_side_effect` in `tests/translator/test_translator.py` to test for side effects between translator executions

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mndeveci — APPROVED** [review]
LGTM

File: tests/translator/test_translator.py [hunk 1]
```diff
@@ -599,6 +599,49 @@ def test_transform_success_resource_policy(self, testcase, partition_with_region
...
+        testcase : Tuple
+            Test name (unused) and Templates
+        templates : List
+            List of templates to transform
+        """
+        partition = partition_with_region[0]
+        region = partition_with_region[1]
+
+        for template in testcase[1]:
+            print(template, partition, region)
```
Comment (wchengru) [inline tests/translator/test_translator.py:639]: Is that a debug left?

Comment (mgrandis) [reply tests/translator/test_translator.py:639]: This is useful when a test fails and you want to know which template/partition/region was responsible.
We already print the output of the transform in `_compare_transform` but it doesn't show the name of the template.
Those are only printed if the test fails.

Comment (wchengru) [reply tests/translator/test_translator.py:639]: Sounds good! I would also recommend to format this information better.

**Review by wchengru — APPROVED** [review]

**Pull request merged**